src/main/java/friday/storage/Storage.java

```diff
@@ -0,0 +1,86 @@
+package friday.storage;
+
+import friday.task.Deadline;
+import friday.task.Event;
+import friday.task.Task;
+import friday.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    private final Path savePath;
+
+    public Storage(Path path) {
+        this.savePath = path;
+    }
+
+    public List<Task> load() {
+        List<Task> list = new ArrayList<>();
+        try {
+            if (!Files.exists(savePath)) {
+                if (savePath.getParent() != null) Files.createDirectories(savePath.getParent());
+                return list;
+            }
+            try (BufferedReader br = Files.newBufferedReader(savePath, StandardCharsets.UTF_8)) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    String[] parts = line.split("\\|", -1);
+                    if (parts.length < 3) continue;
+                    String type = parts[0].trim();
+                    boolean done = "1".equals(parts[1].trim());
+                    String desc = parts[2].trim();
+                    Task t = null;
+                    if ("T".equals(type)) {
+                        t = new Todo(desc);
+                    } else if ("D".equals(type) && parts.length >= 4) {
+                        try {
+                            t = new Deadline(desc, parts[3].trim());
+                        } catch (java.time.format.DateTimeParseException ex) {
+                            t = null;
+                        }
+                    } else if ("E".equals(type) && parts.length >= 5) {
+                        t = new Event(desc, parts[3].trim(), parts[4].trim());
+                    }
+                    if (t != null) {
+                        if (done) t.markAsDone();
+                        list.add(t);
+                    }
+                }
+            }
+        } catch (IOException e) {
+        }
+        return list;
+    }
+
+    public void save(List<Task> tasks) {
+        try {
+            if (savePath.getParent() != null) Files.createDirectories(savePath.getParent());
+            try (BufferedWriter bw = Files.newBufferedWriter(savePath, StandardCharsets.UTF_8)) {
```

Comment:
Nit: Same thing here, would it be better to not abbreviate this?

```suggestion
            try (BufferedWriter writer = Files.newBufferedWriter(savePath, StandardCharsets.UTF_8)) {
```